qubes/ext/vm_config.py

```diff
@@ -0,0 +1,85 @@
+# -*- encoding: utf-8 -*-
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2017 Marek Marczykowski-Górecki
+#                               <marmarek@invisiblethingslab.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see <https://www.gnu.org/licenses/>.
+
+"""Extension exposing vm-config features to QubesDB"""
+
+import qubes.ext
+import qubes.config
+
+PREFIX = 'vm-config.'
+
+
+class VMConfig(qubes.ext.Extension):
+    """This extension export features with 'vm-config.' prefix to QubesDB in
+    /qubes-vm-config/ tree.
```

Comment:
I'd drop `qubes-` part, qubesdb paths have quite low length limit, so lets leave more space for the actual name.